inc/class-addon-manager.php

```diff
@@ -237,6 +238,86 @@ public function has_installed_addons() {
 		return ! empty( $installed_addons );
 	}
 
+	/**
+	 * Checks if the plugin is installed and activated in WordPress.
+	 *
+	 * @param string $slug The class' slug.
+	 *
+	 * @return bool True when installed and activated.
+	 */
+	public function is_installed( $slug ) {
+		$slug_to_class_map = [
+			static::PREMIUM_SLUG     => 'WPSEO_Premium',
+			static::NEWS_SLUG        => 'WPSEO_News',
+			static::WOOCOMMERCE_SLUG => 'Yoast_WooCommerce_SEO',
+			static::VIDEO_SLUG       => 'WPSEO_Video_Sitemap',
+			static::LOCAL_SLUG       => 'WPSEO_Local_Core',
+		];
+
+		if ( ! isset( $slug_to_class_map[ $slug ] ) ) {
+			return false;
+		}
+
+		return class_exists( $slug_to_class_map[ $slug ] );
+	}
+
+	/**
+	 * Validates the addons and show a notice for the ones that are invalid.
+	 */
+	public function validate_addons() {
+		$notification_center = Yoast_Notification_Center::get();
+
+		if ( $notification_center === null ) {
+			return;
+		}
+
+		$addons = [
+			'Yoast SEO Premium'     => static::PREMIUM_SLUG,
+			'News SEO'              => static::NEWS_SLUG,
+			'Yoast WooCommerce SEO' => static::WOOCOMMERCE_SLUG,
+			'Video SEO'             => static::VIDEO_SLUG,
+			'Local SEO'             => static::LOCAL_SLUG,
+		];
+
+		foreach ( $addons as $product_name => $slug ) {
+			$notification = $this->create_notification( $product_name );
```

Comment:
Maybe add this within the if statement so it's only done when needed?

Reply:
We also need this to remove the notification which will be called when condition the if-statement isn't met.